chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -687,13 +687,17 @@ trait TransformerMacros extends TransformerConfiguration with MappingMacros with
       silent = true,
       mode = c.TYPEmode,
       withImplicitViewsDisabled = true,
-      withMacrosDisabled = true
+      withMacrosDisabled = false
     )
 
     scala.util
-      .Try(c.inferImplicitValue(tpeTree.tpe, silent = true, withMacrosDisabled = true))
+      .Try(c.inferImplicitValue(tpeTree.tpe, silent = true, withMacrosDisabled = false))
       .toOption
-      .filterNot(_ == EmptyTree)
+      .filterNot { tree =>
+        tree == EmptyTree ||
+        tree.toString.contains("Transformer.derive") ||
```

Comment:
You have just blown my mind 💥 I'm glad that we have such a simple workaround that seems to cover all the current test suites 👍 

I'm just afraid that when user defines:
```scala
object Transformer {
  implicit val derive: Transformer[...] = ...
}
```
it won't be selected by the library, even if there are no macros involved. That's just only one of many consequences of relying on string filtering.

More reliable method of preventing generation of code that self-references, would be to keep a compile-time stack in `ThreadLocal`, which contains type pairs for which we run our own macro. At the macro invocation we can just lookup the stack and refuse to generate an instance, when there is a type pair on the stack. Quite similar to what [magnolia does](https://github.com/propensive/magnolia/blob/e10e0b153501664987ed07316d150070555e6b4d/core/shared/src/main/scala/magnolia.scala#L801).

Reply:
@krzemin, I found solution without string-based checking. Exactly targeted to to derive methods of io.scalaland.chimney.Transformer and io.scalaland.chimney.TransformerF companions

Reply:
Good 👍 That should be sufficiently reliable IMO.

On the other side, we still perform implicit macro expansion when looking for implicit, so it may cause some performance regression. But it may be fixed later, if needed.

Reply:
Maybe add flag withMacrosDisabled in config with default false value? Users who will not use this feature will be able to disable it and safe compile performance